### PR TITLE
Fix WebAssembly builds for bundlers and browsers

### LINKED ISSUE #126: Using webpack with `p2panda-js` (v0.2.1) is not working "out of the box"
When having `p2panda-js` (v0.2.1) as a dependency in a project using Webpack as a bundler (for example `beep-boop`), the app breaks stating that it could not find the `.wasm` file. Webpack did not consider it since it is loaded via `fetch` inside the `index.min.js` file (this is the file created for browser targets in our build process of `p2panda-js`) and is therefore missing in the dependency tree.

## Possible Solution

To solve the issue with Webpack not importing the `.wasm` file of `p2panda-js` we just found a solution (not elegant, but working for now):

1. Install the CopyWebpackPlugin as a dev-dependency
2. Add the following config under `plugins` in the webpack file (also import the CopyWebpackPlugin):

```js
new CopyPlugin({
  patterns: [
    { from: 'node_modules/p2panda-js/lib/*.wasm', to: '[name][ext]' },
  ],
}),
```

3. Webpack will now copy the needed `.wasm` file into the `dist` folder.

## Another possible solution

Roll back this change: https://github.com/p2panda/p2panda/pull/105 and use wasm inside the bundled file as an base64 encoded string again ..

### PR DESCRIPTION
**Pull request by adzialocha**
**This PR ..**

* Fixes support of `p2panda-js` in Browsers and NodeJS (#126 
* Exports TypeScript definitions (#24) 
* Optimizes WebAssembly build for size and speed
* Requires now `wasm-bindgen` and `wasm-opt` instead of `wasm-pack` to be installed

**New builds**

It changes `p2panda-js` to export three build targets:

1. `node` (for NodeJS environments, no async initialization required)
    ```javascript
    const p2panda = require('p2panda');
    const keyPair = p2panda.createKeyPair();
    console.log(keyPair.publicKey());
    ```
2. `web` (running in Browser and Web-Apps, async one-time initialization required, developers need to manually serve and import `.wasm` file, but faster initialization and smaller files)
    ```javascript
    const { initWebAssembly, createKeyPair } = p2panda;
    await initWebAssembly('/path/to/p2panda.wasm');
    const keyPair = createKeyPair();
    console.log(keyPair.publicKey());
    ```
3. `inline` (running in Browsers and Web-Apps, very easy to get started, nice for quick hacks, async one-time initialization required, larger files)
    ```javascript
    const { initWebAssembly, createKeyPair } = p2panda;
    await initWebAssembly();
    const keyPair = createKeyPair();
    console.log(keyPair.publicKey());
    ```

Closes #126 and #24 (our oldest task 👴)

## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header
- [x] Check if descriptions and terminology match `handbook` content (and visa-versa)

### CODE REVIEW

File: p2panda-js/scripts/compile-wasm.sh [hunk 1]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (cafca) [inline p2panda-js/scripts/compile-wasm.sh:1]: I think this is the most usable non-trivial shell script I have ever seen 👏

File: p2panda-js/README.md [hunk 2]
```diff
@@ -108,15 +152,32 @@ const Chat = ({}) => {
 };
 ```
 
-## Development Setup
+### Manually load `.wasm`
```
Comment (sandreae) [inline p2panda-js/README.md:155]: Very cool we offer this as well :+1:

And these multiple examples are great to have, very developer friendly!

File: p2panda-js/scripts/compile-wasm.sh (hunk 1 above)
Comment (sandreae) [reply p2panda-js/scripts/compile-wasm.sh:1]: wow, nice script!

File: p2panda-js/src/index.ts [hunk 3]
```diff
@@ -3,14 +3,20 @@
 import fetch from 'node-fetch';
 import Headers from 'fetch-headers';
 
+import { setWasmPanicHook } from '../wasm/node';
+
+// @TODO: Remove `fetch` together with `apollo/client`
```
Comment (sandreae) [inline p2panda-js/src/index.ts:8]: :+1:

File: p2panda-js/src/wasm.test.ts [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
+} from './wasm';
 
 const TEST_HASH =
   '0020ddc99aca776df0ca9d1b5871ba39d4edacc752a0a3426b12c3958971b6c847ac';
 const TEST_SCHEMA_ID = `test_${TEST_HASH}`;
 
 describe('WebAssembly interface', () => {
   describe('KeyPair', () => {
-    it('creates a key pair', async () => {
-      const { KeyPair } = await wasm;
```
Comment (sandreae) [inline p2panda-js/src/wasm.test.ts:12]: Nice without `await` everywhere

**Review by sandreae — APPROVED** [review]
This looks great to me! Thanks for doing all this important work :+1:

I really like having the different import methods available and well documented.

**Comment by cafca** [comment]
This is working?? 😍😍 can't wait to see it.

**Pull request merged**